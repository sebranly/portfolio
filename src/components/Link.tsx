import * as React from 'react';
import { useTranslation } from 'react-i18next';

export interface LinkProps {
  keyword: string;
  link: string;
  sectionKeyword: string;
  title: string;
}

const Link: React.FC<LinkProps> = (props) => {
  const { t } = useTranslation();
  const { keyword, link, sectionKeyword, title } = props;
  const tPrefix = 'projects.general';
  const altText = `${t(`${tPrefix}.altText.${sectionKeyword}`)} ${t(title)}`;

  return (
    <div className="inline">
          <div className="font-bold inline">
            {t(`${tPrefix}.${sectionKeyword}`)}
            {t(`${tPrefix}.colon`)}{' '}
          </div>
          <a className="underline inline" href={ link} rel="noopener noreferrer" title={altText} target="_blank">
            {t(`${tPrefix}.${keyword}`)}
          </a>
        </div>
  )
};

export { Link };
