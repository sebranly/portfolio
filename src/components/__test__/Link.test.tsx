import * as React from 'react';
import { render } from '@testing-library/react';

import { Link } from '../Link';

test('Link', () => {
  const github = 'some github link';
  const title = 'some title';
  const createComponent = () => <Link keyword='github' link={github} sectionKeyword='code' title={title} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
