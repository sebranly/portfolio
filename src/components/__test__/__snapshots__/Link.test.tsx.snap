// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link 1`] = `
NodeList [
  <div
    class="inline"
  >
    <div
      class="font-bold inline"
    >
      projects.general.code
      projects.general.colon
       
    </div>
    <a
      class="underline inline"
      href="some github link"
      rel="noopener noreferrer"
      target="_blank"
      title="projects.general.altText.code some title"
    >
      projects.general.github
    </a>
  </div>,
]
`;
